### PR TITLE
Address build warnings and suggestions

### PR DESCRIPTION
**Pull request by nwithan8**
I analyzed the code via my IDE and addressed all warning and suggestion, either by patching it or, in certain cases, purposefully overriding the warning (mostly for linting decisions)

All tests pass as expected, suggesting that no changes affect functionality.

### CODE REVIEW

File: EasyVCR/Censors.cs [hunk 1]
```diff
@@ -238,16 +243,16 @@ public static string CensorJsonData(string data, string censorText, IReadOnlyCol
             {
                 var jsonDictionary = JsonSerialization.ConvertJsonToObject<Dictionary<string, object>>(data);
                 var censoredJsonDictionary = ApplyDataCensors(jsonDictionary, censorText, elementsToCensors);
-                return censoredJsonDictionary == null ? data : JsonSerialization.ConvertObjectToJson(censoredJsonDictionary);
+                return JsonSerialization.ConvertObjectToJson(censoredJsonDictionary);
             }
             catch (Exception)
             {
-                // body is not a JSON dictionary
+                // body is not a JSON dictionary
```
Comment (Justintime50) [inline EasyVCR/Censors.cs:250]: Nit: The trailing space seems counterintuitive?

File: EasyVCR/Defaults.cs [hunk 2]
```diff
@@ -14,60 +14,34 @@ internal static class Defaults
...
-            {
-                "password"
-            },
-            {
-                "secret"
-            },
-            {
-                "username"
-            }
+            "api_key",
```
Comment (Justintime50) [inline EasyVCR/Defaults.cs:33]: Thought: Ooo this is lovely, can we sort this?

**Review by Justintime50 — COMMENTED** [review]
There's a bunch of "ignore this, ignore that" in this PR, I would have hoped for more corrections than ignores. Typically they serve to warn of a missing best practice. Good code won't be easy to write, ignoring is obviously easier but if we ignore most things then what's the point of the error/warnings in the first place?

Is there room here for improvement? I'd expect ignoring things to only be around for something that absolutely cannot happen, a linting mistake made by the tool where the human has better context, and not to simply sweep something under the rug.

**Comment by nwithan8** [comment]
> There's a bunch of "ignore this, ignore that" in this PR, I would have hoped for more corrections than ignores. Typically they serve to warn of a missing best practice. Good code won't be easy to write, ignoring is obviously easier but if we ignore most things then what's the point of the error/warnings in the first place?
> 
> Is there room here for improvement? I'd expect ignoring things to only be around for something that absolutely cannot happen, a linting mistake made by the tool where the human has better context, and not to simply sweep something under the rug.

A lot of these ignores are overrides on the IDE's suggestions for styling. For example, the IDE wants it to be called "Vcr" since title case is the standard in C#. However, we WANT to call it "VCR", so I'm purposely overriding the suggestion.

Others are warnings that certain functions are not being used, since if they are not being used in our test suite, they aren't going to be used at all, at least inside this application (we should expand our code coverage). But just because a method isn't being used inside our test suite doesn't mean we don't want it to be available to the end-user. So we're suppressing the warnings of unused methods. This also applies to some getters/setters that, while not used by our test suite, shouldn't be removed for the end-user's sake.

File: EasyVCR/Defaults.cs (hunk 2 above)
Comment (nwithan8) [reply EasyVCR/Defaults.cs]: Resolved

File: EasyVCR/Censors.cs (hunk 1 above)
Comment (nwithan8) [reply EasyVCR/Censors.cs]: Resolved

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**